tests/test_pipeline/test_mixins.py

```diff
@@ -0,0 +1,205 @@
+    initial_ts = deepcopy(ts)
+    model = MagicMock()
+    model.context_size = context_size
+    mixin = make_mixin(ts=ts, transforms=transforms, model=model, mock_recreate_ts=False)
+
+    ts.fit_transform(transforms)
+
+    start_timestamp = ts.index[start_idx]
+    end_timestamp = ts.index[end_idx]
+    recreated_ts = mixin._recreate_ts(start_timestamp=start_timestamp, end_timestamp=end_timestamp)
+
+    expected_start_timestamp = ts.index[max(0, start_idx - model.context_size)]
+    assert recreated_ts.index[0] == expected_start_timestamp
+    assert recreated_ts.index[-1] == end_timestamp
+    assert recreated_ts.regressors == initial_ts.regressors
+    pd.testing.assert_frame_equal(recreated_ts.df, initial_ts.df[expected_start_timestamp:end_timestamp])
+
+
+@pytest.mark.parametrize("start_idx, end_idx", [(0, 0), (0, 1), (0, 10), (5, 10)])
+def test_determine_prediction_size(start_idx, end_idx, example_tsds):
+    ts = example_tsds
+    mixin = make_mixin(ts=ts, mock_determine_prediction_size=False)
+
+    start_timestamp = ts.index[start_idx]
+    end_timestamp = ts.index[end_idx]
+    prediction_size = mixin._determine_prediction_size(start_timestamp=start_timestamp, end_timestamp=end_timestamp)
+
+    expected_prediction_size = end_idx - start_idx + 1
+    assert prediction_size == expected_prediction_size
+
+
+@pytest.mark.parametrize("start_idx, end_idx", [(0, 0), (0, 1), (0, 10), (5, 10)])
+def test_predict_recreate_ts_called(start_idx, end_idx, example_tsds):
+    mixin = make_mixin()
+
+    start_timestamp = mixin.ts.index[start_idx]
+    end_timestamp = mixin.ts.index[end_idx]
+    _ = mixin._predict(
+        start_timestamp=start_timestamp, end_timestamp=end_timestamp, prediction_interval=False, quantiles=[]
+    )
+
+    mixin._recreate_ts.assert_called_once_with(start_timestamp=start_timestamp, end_timestamp=end_timestamp)
+
+
+@pytest.mark.parametrize("start_idx, end_idx", [(0, 0), (0, 1), (0, 10), (5, 10)])
+def test_predict_determine_prediction_size_called(start_idx, end_idx, example_tsds):
+    mixin = make_mixin()
+
+    start_timestamp = mixin.ts.index[start_idx]
+    end_timestamp = mixin.ts.index[end_idx]
+    _ = mixin._predict(
+        start_timestamp=start_timestamp, end_timestamp=end_timestamp, prediction_interval=False, quantiles=[]
+    )
+
+    mixin._determine_prediction_size.assert_called_once_with(
+        start_timestamp=start_timestamp, end_timestamp=end_timestamp
+    )
+
+
+@pytest.mark.parametrize(
+    "transforms",
+    [
+        [DateFlagsTransform()],
+        [FilterFeaturesTransform(exclude=["regressor_exog_weekend"])],
+    ],
+)
+def test_predict_apply_transforms(transforms):
+    mixin = make_mixin(transforms=transforms)
+
+    _ = mixin._predict(
+        start_timestamp=pd.Timestamp("2020-01-01"),
+        end_timestamp=pd.Timestamp("2020-01-02"),
+        prediction_interval=False,
+        quantiles=[],
+    )
+
+    expected_ts = mixin._recreate_ts.return_value
```

Comment:
exprcted_ts from return_value?

Reply:
Method `_recreate_ts` is tested. So in this test I check that on the result of calling `_recreate_ts` the transforms are applied as expected.

Reply:
It doesn't make more sense to me.But it's minor issue that could be ignored